source/material_model/steinberger.cc

```diff
@@ -270,6 +280,27 @@ namespace aspect
 
 
 
+    template <int dim>
+    void
+    Steinberger<dim>::
+    fill_prescribed_outputs(const unsigned int q,
+                            const std::vector<double> &,
+                            const MaterialModel::MaterialModelInputs<dim> &,
+                            MaterialModel::MaterialModelOutputs<dim> &out) const
+    {
+      // set up variable to interpolate prescribed field outputs onto compositional field
+      PrescribedFieldOutputs<dim> *prescribed_field_out = out.template get_additional_output<PrescribedFieldOutputs<dim> >();
+
+      if (this->introspection().compositional_name_exists("density_field")
```

Comment:
Would it make sense to introduce ``CompositionalFieldDescription::density_field`` ? Identification by name is no longer necessary now that we have this system...